sdk/core/core/inc/az_span.h

```diff
@@ -101,12 +113,69 @@ AZ_INLINE bool az_const_span_eq(az_const_span const a, az_const_span const b) {
   return a.size == b.size && memcmp(a.begin, b.begin, a.size) == 0;
 }
 
-AZ_INLINE az_result az_span_copy(az_const_span const src, az_span const dst) {
-  if (dst.size < src.size) {
+AZ_INLINE az_result
+az_span_move(az_span const buffer, az_const_span const src, az_span * const out_result) {
+  AZ_CONTRACT_ARG_NOT_NULL(out_result);
+  if (!az_span_is_valid(buffer) || !az_const_span_is_valid(src)) {
+    return AZ_ERROR_ARG;
+  }
+
+  if (buffer.size < src.size) {
     return AZ_ERROR_BUFFER_OVERFLOW;
   }
 
-  memcpy(dst.begin, src.begin, src.size);
+  if (!az_const_span_is_empty(src)) {
+    memmove((void *)buffer.begin, (void const *)src.begin, src.size * sizeof(*src.begin));
+  }
+
+  out_result->begin = buffer.begin;
+  out_result->size = src.size;
+
+  return AZ_OK;
+}
+
+AZ_INLINE uint8_t * az_span_end(az_span const span) { return &span.begin[span.size]; }
```

Comment:
`span.begin + span.size`. I would avoid `span.begin[span.size]` even if it's temporary.